WiFiManager.cpp

```diff
@@ -201,50 +255,47 @@ boolean  WiFiManager::startConfigPortal(char const *apName, char const *apPasswo
         break;
       }
     }
+    if (stopConfigPortal) {
+	  stopConfigPortal = false;
+	  break;
+    }
     yield();
   }
-
+  WiFi.mode(WIFI_STA);
+  if (TimedOut & WiFi.status() != WL_CONNECTED) {
+	WiFi.begin();
+    int connRes = waitForConnectResult();
+    DEBUG_WM ("Timed out connection result: ");
+    DEBUG_WM ( getStatus(connRes));
+    }
   server.reset();
   dnsServer.reset();
-
   return  WiFi.status() == WL_CONNECTED;
 }
 
-
 int WiFiManager::connectWifi(String ssid, String pass) {
-  DEBUG_WM(F("Connecting as wifi client..."));
-
+  DEBUG_WM(F("Connecting wifi with new parameters..."));
+  if (ssid != "") {
+	resetSettings(); /*Disconnect and wipe out old values. If you don't do this
```

Comment:
resetSettings is only called when a new SSID value has been provided because of the line - if (ssid != "")

Therefore previous settings remain unchanged if new ones are not provided.